packages/expo-cli/src/commands/build/BaseBuilder.js

```diff
@@ -89,6 +89,13 @@ export default class BaseBuilder {
       log.error(`'expo build:${this.platform()}' is not supported for detached projects.`);
       process.exit(1);
     }
+    
+    // Warn user if building a project using the next deprecated SDK version
+    let nextDeprecatedVersion = await Versions.nextDeprecatedSdkVersionAsync();
+    if (this.manifest.sdkVersion.substring(0, 2) === nextDeprecatedVersion.toString()) {
```

Comment:
Please use `server.major(this.manifest.sdkVersion)` to parse the major version (it doesn't assume the major version only has 2 digits).